consensus/src/quorum_store/utils.rs

```diff
@@ -486,6 +482,8 @@ impl ProofQueue {
                 }
             }
         }
+        self.txn_summary_num_occurrences
```

Comment:
Might be faster to remove right away `entry = txn_summary_num_occurrences.entry()` and `if count == 0 entry.remove()` to avoid a linear scan.

But seems like the update is now fast enough anyway, so if you find this is easier to understand then that's fine.